core/shape-parts.cpp

```diff
@@ -39,6 +39,32 @@ const Path& BasicBezier::generate(Point endPoint, Point transitionPoint) {
   return m_path;
 }
 
+const Path& FlatBezier::generate(Point endPoint, Point transitionPoint) {
+  MCFixed flatEndY = m_flatWidth / -2; // WHY DO I HAVE TO MAKE THIS NEGATIVE TO GET IT TO WORK???
+  if (endPoint.X < transitionPoint.X) {
+    Path ep;
+    ep.push_back(endPoint);
+    m_path = BezierPath(Point(endPoint.X, flatEndY),
+                        Point(endPoint.X, flatEndY + 
```

Comment:
I started out by making this cover the entire space but by just covering the distance from the end of the flat section it maintains a consistent shape of the bezier curve whatever the size of the flat section. Also it means all values of end handle between 0 and 1 are still valid (which they wouldn't be if it scales across the entire space). I can change it if you like but I think I prefer the way I have it at the moment.